sklearn/neighbors/tests/test_neighbors.py

```diff
@@ -1315,59 +1429,26 @@ def test_neighbors_badargs():
         nbrs.radius_neighbors_graph(X, mode="blah")
 
 
-# TODO: Remove filterwarnings in 1.3 when wminkowski is removed
 @pytest.mark.filterwarnings("ignore:WMinkowskiDistance:FutureWarning:sklearn")
-def test_neighbors_metrics(n_samples=20, n_features=3, n_query_pts=2, n_neighbors=5):
+@pytest.mark.parametrize("metric", COMMON_VALID_METRICS)
```

Comment:
Ideally, this should check for all metrics supported by `"ball_tree"` (minus `"pyfunc"`) to make sure that all the metrics supported by "ball_tree"` yield the same results as the brute force.

The subset of metrics that are not supported by `"kd_tree"` should be tested to raise `ValueError` explicitly when using the `"kd_tree"`.